os/net/mac/csma/csma.c

```diff
@@ -91,20 +87,11 @@ input_packet(void)
     }
 
 #if CSMA_SEND_SOFT_ACK
-    {
-      frame802154_t info154;
-      frame802154_parse(original_dataptr, original_datalen, &info154);
-      if(info154.fcf.frame_type == FRAME802154_DATAFRAME &&
-         info154.fcf.ack_required != 0 &&
-         linkaddr_cmp((linkaddr_t *)&info154.dest_addr,
-                      &linkaddr_node_addr)) {
-        uint8_t ackdata[CSMA_ACK_LEN] = {0, 0, 0};
-
-        ackdata[0] = FRAME802154_ACKFRAME;
-        ackdata[1] = 0;
-        ackdata[2] = info154.seq;
-        NETSTACK_RADIO.send(ackdata, CSMA_ACK_LEN);
-      }
+    if(packetbuf_attr(PACKETBUF_ATTR_MAC_ACK)) {
```

Comment:
in the case where `if(packetbuf_datalen() == CSMA_ACK_LEN)`, isn't there a risk that the frame is not parsed yet, and the packetbuf attribute here is just a leftover from last in or out frame?

Reply:
No. That if is in the very last else branch, which is only executed if packetbuf_datalen() != CSMA_ACK_LEN and parsing was successful.

Reply:
Got it, I was somehow looking at the wrong indent level. Agree.